packages/react-core/src/modules/in-app-message/hooks/useInAppMessages.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  FetchInAppMessagesOptions,
+  InAppMessage,
+  InAppMessageClient,
+} from "@knocklabs/client";
+import { useStore } from "@tanstack/react-store";
+import { useEffect, useMemo } from "react";
+
+import { useInAppMessageChannel } from "../context";
+
+// TODO: Type so that the message content (or fields) is typed
+export const useInAppMessages = (
+  messageType: string,
+  options: FetchInAppMessagesOptions = {},
+) => {
+  const { inAppChannelClient } = useInAppMessageChannel();
+
+  const inAppMessageClient = useMemo(() => {
+    return new InAppMessageClient(inAppChannelClient, messageType, options);
+  }, [inAppChannelClient, messageType, options]);
+
+  // TODO: Is it possible to extract all store usage?
+  // Maybe not because framework specific hooks need to be a called to support reactivity
+  // Could definitely create selectors as functions in the client library though
+  const messages = useStore(inAppChannelClient.store, (state) => {
+    const messageIds = new Set(
+      state.queries[inAppMessageClient.queryKey]?.data?.items?.map(
+        (message) => message.id,
+      ),
+    );
+
+    const messages = Object.entries(state.messages).reduce<InAppMessage[]>(
```

Comment:
To thomas' point above, this seems wrong right now. I think we should be able to say:

```typescript
const messages = messageIds.map(mid => state.messages[mid])
```

Reply:
Store lookup is O(1), and then complexity is O(N) to number of messages in result set, not O(N) for messages in store.

Reply:
Yeah, will clean that up!